ui/admin/tests/acceptance/roles/global-scope-test.js

```diff
@@ -0,0 +1,494 @@
+    roles: null,
+    role: null,
+    roleScopes: null,
+    manageScopes: null,
+    manageCustomScopes: null,
+  };
+
+  hooks.beforeEach(function () {
+    authenticateSession({});
+    instances.scopes.global = this.server.create('scope', { id: 'global' });
+    instances.scopes.org = this.server.create('scope', {
+      type: 'org',
+      scope: { id: 'global', type: 'global' },
+    });
+    instances.scopes.project = this.server.create('scope', {
+      type: 'project',
+      scope: { id: instances.scopes.org.id, type: 'org' },
+    });
+    instances.role = this.server.create(
+      'role',
+      {
+        scope: instances.scopes.global,
+      },
+      'withScopes',
+    );
+
+    urls.roles = `/scopes/global/roles`;
+    urls.role = `${urls.roles}/${instances.role.id}`;
+    urls.roleScopes = `${urls.role}/scopes`;
+    urls.manageScopes = `${urls.role}/manage-scopes`;
+    urls.manageCustomScopes = `${urls.manageScopes}/manage-custom-scopes`;
+  });
+
+  test('visiting role scopes', async function (assert) {
+    await visit(urls.role);
+
+    await click(`[href="${urls.roleScopes}"]`);
+    await a11yAudit();
+
+    assert.strictEqual(currentURL(), urls.roleScopes);
+    assert
+      .dom(TABLE_ROW_SELECTOR)
+      .exists({ count: instances.role.grant_scope_ids.length });
+  });
+
+  test('user can naviage to scope from role grant scopes', async function (assert) {
+    await visit(urls.role);
+
+    await click(`[href="${urls.roleScopes}"]`);
+    await click('tbody tr:nth-child(2) a');
+
+    assert.strictEqual(
+      currentURL(),
+      `/scopes/${instances.scopes.global.id}/scopes`,
+    );
+  });
+
+  test('user can naviage to parent scope from role grant scopes', async function (assert) {
+    instances.role.update({
+      grant_scope_ids: ['this', instances.scopes.org.id],
+    });
+    await visit(urls.role);
+
+    await click(`[href="${urls.roleScopes}"]`);
+    await click('tbody tr:nth-child(2) td:nth-child(3) a');
+
+    assert.strictEqual(
+      currentURL(),
+      `/scopes/${instances.scopes.global.id}/scopes`,
+    );
+  });
+
+  test('user sees no scopes message and action when role has no grant scopes', async function (assert) {
+    instances.role.update({ grant_scope_ids: [] });
+    await visit(urls.role);
+
+    await click(`[href="${urls.roleScopes}"]`);
+
+    assert.dom('.role-grant-scopes div').includesText('No scopes added');
+    assert.dom('.role-grant-scopes div div:nth-child(3) a').isVisible();
+  });
+
+  test('user does not see action to add scopes when role has no grant scopes without proper permissions', async function (assert) {
+    instances.role.update({
+      grant_scope_ids: [],
+      authorized_actions: instances.role.authorized_actions.filter(
+        (action) => action !== 'set-grant-scopes',
+      ),
+    });
+    await visit(urls.role);
+
+    await click(`[href="${urls.roleScopes}"]`);
+
+    assert.dom('.role-grant-scopes div div:nth-child(3) a').doesNotExist();
```

Comment:
Could we move this (and other complex selectors) to a const to help us know what we are asserting does not exist?

Reply:
good catch, this was updated, thank you!